src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -171,4 +177,34 @@ void GIVEN_existing_sync_information_WHEN_config_updates_THEN_removed_sync_infor
                 new Pair<>(THING_NAME2, "Shadow-0"),
                 new Pair<>(THING_NAME2, "Shadow-5")));
     }
+
+
+    @Test
+    @SuppressWarnings("PMD.CloseResource")
+    void GIVEN_shadow_manager_WHEN_individual_config_resets_THEN_respond_to_config_updates(ExtensionContext context)
+            throws Exception {
+        ignoreExceptionOfType(context, SkipSyncRequestException.class);
+        MqttClient mqttClient = mock(MqttClient.class);
+        lenient().when(mqttClient.connected()).thenReturn(false);
+
+
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
+                .configFile(DEFAULT_CONFIG)
+                .mqttConnected(false)
+                .build());
+        SyncHandler syncHandler = kernel.getContext().get(SyncHandler.class);
+        assertThat(syncHandler.getSyncDirection(), is(Direction.BETWEEN_DEVICE_AND_CLOUD));
+
+        shadowManager.getConfig().lookupTopics(CONFIGURATION_CONFIG_KEY, CONFIGURATION_SYNCHRONIZATION_TOPIC)
```

Comment:
Mmm the `DEFAULT_CONFIG` doesn't specify any syncConfiguration so removing it seems kind of weird since it wasn't there in the first place. It would be nice if the initial config was `DEVICE_TO_CLOUD` then you remove it and then changes to `BETWEEN_DEVICE_AND_CLOUD` and then change it again to `DEVICE_TO_CLOUD` that way we know

1) It sets the correct value based on the original configuration (use non default value)
2) It responds correctly when removing a value by reverting back to its default value
3) When there is no existing value and you add one then it updates it to the value you want

Reply:
In the code, new node is added with the default value so we're able to remove. But yes, I agree with your comment, so updated the PR.